pkg/postgres/postgres.go

```diff
@@ -0,0 +1,90 @@
+package postgres
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/lib/pq"
+)
+
+func Connect(log logr.Logger, connectionString string) (*sql.DB, error) {
+	log.Info("Connecting to database", "url", connectionString)
+	db, err := sql.Open("postgres", connectionString)
+	if err != nil {
+		return nil, err
+	}
+	err = db.Ping()
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+type Privilege int
+
+const (
+	PrivilegeRead  Privilege = iota
+	PrivilegeWrite Privilege = iota
+)
+
+type DatabaseSchema struct {
+	Name       string
+	Schema     string
+	Privileges Privilege
+}
+
+func Role(log logr.Logger, db *sql.DB, name string, roles []string, databases []DatabaseSchema) error {
+	log.Info(fmt.Sprintf("Creating role %s", name))
+	query := fmt.Sprintf("CREATE ROLE %s WITH LOGIN", name)
+	if len(roles) != 0 {
+		query += fmt.Sprintf(" IN ROLE %s", strings.Join(roles, ", "))
+	}
+	_, err := db.Exec(query)
+	if err != nil {
+		pqError, ok := err.(*pq.Error)
+		if !ok || pqError.Code.Name() != "duplicate_object" {
+			return err
+		}
+		log.Info(fmt.Sprintf("Role %s already exists", name), "errorCode", pqError.Code, "errorName", pqError.Code.Name())
+	} else {
+		log.Info(fmt.Sprintf("Role %s created", name))
+	}
+	if len(roles) != 0 {
+		_, err = db.Exec(fmt.Sprintf("GRANT %s TO %s", strings.Join(roles, ", "), name))
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, database := range databases {
+		// Only needed for testing without rds_iam role that otherwise grants this right
+		log.Info(fmt.Sprintf("Granting CONNECT to database '%s'", database.Name))
+		_, err = db.Exec(fmt.Sprintf("GRANT CONNECT ON DATABASE %s TO %s", database.Name, name))
+		if err != nil {
+			return fmt.Errorf("grant connect on database '%s': %w", database.Name, err)
+		}
+		log.Info(fmt.Sprintf("Granting USAGE of schema '%s'", database.Name))
+		_, err = db.Exec(fmt.Sprintf("GRANT USAGE ON SCHEMA %s TO %s", database.Name, name))
```

Comment:
Is this given to all developers or just services? 
I just looked at USAGE again and I'm not sure if this is the right one to give: https://www.postgresql.org/docs/10/sql-grant.html

Reply:
This is granted all developers as they need it to to see tables in the schema.

As the PUBLIC role is limited developers cannot edit objects in the schema anyways, as I understand it.